xwiki-commons-core/xwiki-commons-xml/src/test/java/org/xwiki/xml/XMLUtilsTest.java

```diff
@@ -498,10 +498,11 @@ void transformAvoidXMLBomb()
 
         assertEquals(1, logCapture.size());
         assertEquals(Level.WARN, logCapture.getLogEvent(0).getLevel());
-        assertEquals("Failed to apply XSLT transformation: [javax.xml.transform.TransformerException: "
-            + "com.sun.org.apache.xml.internal.utils.WrappedRuntimeException: The parser has encountered more than "
-            + "\"100,000\" entity expansions in this document; this is the limit imposed by the application.]",
-            logCapture.getMessage(0));
+        String txt = "Failed to apply XSLT transformation: \\[javax.xml.transform.TransformerException: "
+                + "com.sun.org.apache.xml.internal.utils.WrappedRuntimeException: The parser has encountered more than "
+                + "\"100.000\" entity expansions in this document; this is the limit imposed by the application.]";
+        assertTrue(logCapture.getMessage(0).matches(txt), () -> "expected : <" +txt.replaceAll("\\\\","")
```

Comment:
I think I'd prefer to use the syntax `assertThat("Linux at Baeldung").matches(".* at Baeldung$");`

Reply:
assertThat isn't part of junit5. I could use AssertJ.aasertThat, but this would introduce a new dependency.

Reply:
It's part of Hamcrest. Example usage in XWiki: https://github.com/xwiki/xwiki-platform/blob/f3ebce137fcce0887f23872971387edfd9a79db0/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/test/java/org/xwiki/web/DocumentTagsTest.java#L111

Now the most important aspect to address is Thomas's comment at https://jira.xwiki.org/browse/XCOMMONS-2858?focusedId=115937&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-115937

Reply:
> Now the most important aspect to address is Thomas's comment at [jira.xwiki.org/browse/XCOMMONS-2858?focusedId=115937&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-115937](https://jira.xwiki.org/browse/XCOMMONS-2858?focusedId=115937&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-115937)

If it's Xerces there is probably not much we can do, unfortunately. I guess we'll try again https://jira.xwiki.org/browse/XCOMMONS-2175 when moving to Java 17, and it's possible the standard Java version of the same protection does not have this bug.